t/e2e/whisper/whisper_test.go

```diff
@@ -176,32 +191,65 @@ func (s *WhisperTestSuite) TestSelectedAccountOnRestart() {
 	s.Require().NoError(s.Backend.StartNode(&preservedNodeConfig))
 
 	// re-check selected account (account2 MUST be selected)
-	selectedAccount, err = s.Backend.AccountManager().SelectedAccount()
+	selectedWalletAccount, err = s.Backend.AccountManager().SelectedWalletAccount()
 	s.NoError(err)
-	s.NotNil(selectedAccount)
-	s.Equal(selectedAccount.Address.Hex(), address2, "incorrect address selected")
+	s.NotNil(selectedWalletAccount)
+	s.Equal(selectedWalletAccount.Address.Hex(), address2, "incorrect address selected")
 
 	// make sure that Whisper gets identity re-injected
 	whisperService = s.WhisperService()
-	s.True(whisperService.HasKeyPair(pubKey2), "identity not injected into whisper")
-	s.False(whisperService.HasKeyPair(pubKey1), "identity should not be present, but it is still present in whisper")
+	s.True(whisperService.HasKeyPair(selectedChatPubKey2), "identity not injected into whisper")
+	s.False(whisperService.HasKeyPair(selectedChatPubKey1), "identity should not be present, but it is still present in whisper")
 
 	// now restart node using RestartNode() method, and make sure that account is still available
 	s.RestartTestNode()
 	defer s.StopTestBackend()
 
 	whisperService = s.WhisperService()
-	s.True(whisperService.HasKeyPair(pubKey2), "identity not injected into whisper")
-	s.False(whisperService.HasKeyPair(pubKey1), "identity should not be present, but it is still present in whisper")
+	s.True(whisperService.HasKeyPair(selectedChatPubKey2), "identity not injected into whisper")
+	s.False(whisperService.HasKeyPair(selectedChatPubKey1), "identity should not be present, but it is still present in whisper")
 
 	// now logout, and make sure that on restart no account is selected (i.e. logout works properly)
 	s.NoError(s.Backend.Logout())
 	s.RestartTestNode()
 	whisperService = s.WhisperService()
-	s.False(whisperService.HasKeyPair(pubKey2), "identity not injected into whisper")
-	s.False(whisperService.HasKeyPair(pubKey1), "identity should not be present, but it is still present in whisper")
+	s.False(whisperService.HasKeyPair(selectedChatPubKey2), "identity not injected into whisper")
+	s.False(whisperService.HasKeyPair(selectedChatPubKey1), "identity should not be present, but it is still present in whisper")
+
+	selectedWalletAccount, err = s.Backend.AccountManager().SelectedWalletAccount()
+	s.EqualError(account.ErrNoAccountSelected, err.Error())
+	s.Nil(selectedWalletAccount)
 
-	selectedAccount, err = s.Backend.AccountManager().SelectedAccount()
+	selectedChatAccount, err = s.Backend.AccountManager().SelectedChatAccount()
 	s.EqualError(account.ErrNoAccountSelected, err.Error())
-	s.Nil(selectedAccount)
+	s.Nil(selectedChatAccount)
+}
+
+func (s *WhisperTestSuite) TestWalletAndChatPubKeyAreTheSame() {
```

Comment:
Oh actually, can we put this into `accounts_test.go`? Because this is e2e test so it's not that often executed and this test does not require a running node.

Reply:
@adambabik I added this check in the account test when we call `SelectAccount`.
I would like to leave this one as well (I renamed it to `TestSelectedChatKeyIsUsedInWhisper`), so that we check that we inject the right key. what do you think?